monitoring/scripts/run_comprehensive_tests.sh

```
@@ -0,0 +1,76 @@
+echo "   Test Coverage: Full monitoring system validation"
+echo "   Services: Prometheus, Grafana, Alertmanager, Metrics, Webhook"
+echo ""
+
+# Check if ports are available for testing
+echo "🔍 Checking port availability..."
+ports_to_check=(9090 3001 9093 8000 5001)
+available_ports=()
+
+for port in "${ports_to_check[@]}"; do
+    if ! lsof -Pi :$port -sTCP:LISTEN -t >/dev/null 2>&1; then
+        available_ports+=($port)
```

Shellcheck (0.10.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)